pkg/providers/pricing/pricing.go

```diff
@@ -415,6 +416,14 @@ func (p *Provider) Reset() {
 	p.onDemandPrices = staticPricing
 	// default our spot pricing to the same as the on-demand pricing until a price update
 	p.spotPrices = populateInitialSpotPricing(staticPricing)
-	p.onDemandUpdateTime = initialPriceUpdate
-	p.spotUpdateTime = initialPriceUpdate
+	p.onDemandUpdateTime = getInitialPriceUpdateByRegion(p.region)
+	p.spotUpdateTime = getInitialPriceUpdateByRegion(p.region)
+}
+
+func getInitialPriceUpdateByRegion(region string) time.Time {
```

Comment:
nit: `lo.Ternary` for this?

Reply:
Unless we don't plan on adding `cn-northwest-1` to this, I think this is cleaner than a ternary, especially since this is used in two places.

Reply:
To me, that's over-optimizing a bit for the future rather than just building for now. We don't know if we will or won't. I'm fine either way but I'd generally caution against optimizing a function for something that we _think_ we might do as a philosophy.

Reply:
That's fair. With the change two the 3 AWS partitions vs the 2 unofficial partitions, I think this is quite a bit cleaner so I'll keep this for now.